internal/config/config.go

```diff
@@ -143,5 +174,60 @@ func Validate(filePath string) (Config, error) {
 	config.PebbleNotificationsEnabled = c.PebbleNotifications
 	config.Logging.System.Level = LoggingLevel(c.Logging.System.Level)
 	config.Logging.System.Output = c.Logging.System.Output
+	config.EncryptionBackend, err = createEncryptionBackend(c.EncryptionBackend)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to create encryption backend: %w", err)
+	}
 	return config, nil
 }
+
+// createEncryptionBackend creates a SecretBackend based on the type specified
```

Comment:
It feels wrong for the config code to create an encryption backend. The config file and code is really only meant for fetching and validating config options. I think this function and calling it belongs elsewhere. In other words, `internal/config` should probably not import other `internal/...` packages.